PRINCIPLES.md

```diff
@@ -0,0 +1,186 @@
+
+GitOps principles were derived from modern software operations but are rooted in pre-existing and widely adopted best practices. These principles are:
+
+<!-- Cross-link principles here to the longer discussion notes for each below -->
+1. [**The principle of declarative configuration**](#tag)
+
+    A system managed by GitOps must have its _Desired State_ expressed declaritively as data. 
+    This declarative data must be in a format writable and readable by both humans and software.
+
+2. [**The principle of immutable configuration versions**](#tag)
+
+    _Desired State_ is stored in a way that supports versioning, immutability of versions, and retains a complete version history.
+    We call systems that store desired state in this way _State Stores_.
+
+3. [**The principle of continuous state reconciliation**](#tag)
+
+    Software agents continuously, and automatically, compare a system's _Actual State_ to its _Desired State_.
+    If the actual and desired states differ, automated actions are immediately attempted to reconcile them.
+    These differences could be due to the actual state drifting from the desired state, or the desired state changing intentionally.
+
+4. [**The principle of operations through declaration**](#tag)
+
+    When wishing to operate on a software system, a human or software agent will not interact with the running system and modify it directly.
+    Instead, the agent will create a new declarative version of the desired state in the state store.
+
+## Introduction
+
+The software systems that we manage vary widely; from battery powered embedded systems driven by microcontrollers, to globally distributed applications with millions of users. 
+Most systems are composed of subsystems themselves made of software.
+It is impossible to comprehensively outline all the specific practices for managing such a variety of systems.
+
+However, despite the myriad differences, several important principles emerge that simplify the task of reliably managing and operating _all_ software systems at scale. GitOps attempts to capture some of these principles in a coherent framework.
+The GitOps principles can be applied to managing entire software systems or applied only to parts of larger systems. 
+
+## Scope
+
+GitOps concerns the verifiable behaviour of computer systems and their interfaces. 
+Specifically, GitOps is not about human processes, and is not intended as a model for judging human organisational designs and operational practices.
+
+The GitOps principles are to be used as guiding principles in the development of modern software and system operations. They do not form a concrete specification. 
+
+The GitOps principles are a direction, not a destination. They should be applied pragmatically. For example, whilst they can be applied strictly to an entire software systems, they can also be applied loosely to selected parts of larger systems as part of a progressive adoption.
+
+## The GitOps Principles
```

Comment:
is GitOps the correct term still here? Are we tying ourselves too much to Git (software tool)? Flip side - its common and recognizable.

Reply:
New principal: "No out of process access"

consider adding "how to implement these principals"
* all non-gitops changes are tracked and you should drive this down to 0